companion/src/shared/timeredit.cpp

```diff
@@ -0,0 +1,281 @@
+          ret = match.captured("hrs").toInt();
+        break;
+
+      case Minutes:
+        if (!match.captured("mins").isEmpty())
+          ret = match.captured("mins").toInt();
+        break;
+
+      case Seconds:
+        if (!match.captured("secs").isEmpty())
+          ret = match.captured("secs").toInt();
+        break;
+
+      case Polarity:
+        if (!match.captured("pol").isEmpty() && !match.captured("pol").compare("-"))
+          ret = -1;
+        else
+          ret = 1;
+        break;
+
+      default:
+        break;
+    }
+  }
+  //qDebug() << input << match << ret;
+
+  return ret;
+}
+
+
+//// Slots ////
+
+void TimerEdit::setTime(int seconds)
+{
+  if (seconds > m_maxTime)
+    seconds = m_maxTime;
+  else if (seconds < m_minTime)
+    seconds = m_minTime;
+
+  div_t qrM = div(abs(seconds), 60);
+  div_t qrH = div(qrM.quot, 60);
+
+  QString val = QString("%1").arg(qrH.rem, 2, 10, QChar('0'));
+
+  if (m_hourDigits)
+    val.prepend(QString("%1:").arg(qrH.quot, m_hourDigits, 10, QChar('0')));
+
+  if (m_showSeconds)
+    val.append(QString(":%1").arg(qrM.rem, 2, 10, QChar('0')));
+
+  if (m_minTime < 0) {
+    if (seconds < 0)
+      val.prepend("-");
+    else
+      val.prepend(" ");
+  }
+  //qDebug() << seconds << qrH.quot << qrH.rem << qrM.quot << qrM.rem << val << text();
+
+  if (text().compare(val)) {
+    setText(val);
+  }
+}
+
+void TimerEdit::incrDecr(int seconds)
+{
+  if (!seconds)
+    return;
+
+  const int cs = timeInSeconds();
+  int s = seconds + cs;
+  if (s < m_minTime)
+    s = m_minTime;
+  if (s > m_maxTime)
+    s = m_maxTime;
+  if (s != cs) {
+    setTime(s);
+    emitValueChanged();
+  }
+}
+
+void TimerEdit::setTimeRange(int minSeconds, int maxSeconds)
+{
+  if (minSeconds <= maxSeconds && (m_minTime != minSeconds || m_maxTime != maxSeconds)) {
+    int digits = 0;
+    if (maxSeconds >= 3600)
+      digits = floorf(logf(round(maxSeconds / 3600)) / logf(10.0f)) + 1;
+
+    bool mod = (m_hourDigits != digits || std::signbit((float)m_minTime) != std::signbit((float)minSeconds));
+
+    m_minTime = minSeconds;
+    m_maxTime = maxSeconds;
+    m_hourDigits = digits;
+
+    if (mod)
+      setupFormat();
+  }
+}
+
+void TimerEdit::setMinimumTime(int minSeconds)
+{
+  setTimeRange(minSeconds, (m_maxTime > minSeconds ? m_maxTime : minSeconds));
+}
+
+void TimerEdit::setMaximumTime(int maxSeconds)
+{
+  setTimeRange((m_minTime < maxSeconds ? m_minTime : maxSeconds), maxSeconds);
+}
+
+void TimerEdit::setShowSeconds(bool showSeconds)
+{
+  if (m_showSeconds != showSeconds) {
+    m_showSeconds = showSeconds;
+    setupFormat();
+  }
+}
+
+
+//// Protected ////
+
+void TimerEdit::textEditedHandler()
+{
+  int sec = timeInSeconds();
+  if (sec < m_minTime)
+    setTime(m_minTime);
+  else if (sec > m_maxTime)
+    setTime(m_maxTime);
+  else
+    return;
+
+  emitValueChanged();
+
+  //qDebug() << text() << sec << getTimeSeconds() << validator()->validate(text(), sec);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this block of commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3ACommentedCode)